### PR TITLE
feat(tests): add comprehensive tests for cleanup.py. Closes #699

### LINKED ISSUE #699: Database retention logic untested
Currently our retention logic is untested. This should be changed by adding a comprehensive set of tests that cover `greedybear/cronjobs/cleanup.py`.

### PR DESCRIPTION
**Pull request by opbot-xd**
# Description

This PR implements comprehensive unit tests for [greedybear/cronjobs/cleanup.py](cci:7://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/cronjobs/cleanup.py:0:0-0:0) as requested in issue #699. This logic was previously untested.

The tests ensure that the cleanup job correctly:
- Initializes with default or injected repositories.
- Calcluates the correct expiration dates for IOCs, Cowrie Sessions, and Command Sequences based on settings.
- Calls the appropriate deletion methods on the [IocRepository](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/tests/test_ioc_repository.py:11:0-406:63) and [CowrieSessionRepository](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/tests/test_cowrie_session_repository.py:10:0-102:13) with those dates.
- Handles edge cases like zero deletions correctly.

## Related issues
Closes #699

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/GreedyBear/Contribute/) to this project.
- [x] The pull request is for the branch `develop`.
- [ ] I have added documentation of the new features.
- [x] Linter (`Ruff`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowlproject.github.io/docs/GreedyBear/Contribute/#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] I have added tests for the feature/bug I solved. All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/docs/blob/main/docs/GreedyBear/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### Code Coverage Report
Ran tests for `greedybear/cronjobs/cleanup.py`:
```text
Name                             Stmts   Miss  Cover
----------------------------------------------------
greedybear/cronjobs/cleanup.py      29      0   100%
----------------------------------------------------
TOTAL                               29      0   100%

### CODE REVIEW

**Comment by opbot-xd** [comment]
Method used for coverage:
```bash
docker exec greedybear_uwsgi pip install coverage
docker exec greedybear_uwsgi coverage run --source=greedybear.cronjobs.cleanup manage.py test tests.greedybear.cronjobs.test_cleanup
docker exec greedybear_uwsgi coverage report
```

**Comment by opbot-xd** [comment]
Following the Copilot suggestion, I updated line 78 in [tests/greedybear/cronjobs/test_cleanup.py](cci:7://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/tests/greedybear/cronjobs/test_cleanup.py:0:0-0:0). I replaced the loose `assertTrue(count >= 10)` with a strict `assertEqual(count, 10)` to ensure we are verifying the exact number of expected log messages.

**Comment by opbot-xd** [comment]
### Extended Code Coverage Analysis

I ran a broader coverage check across the `greedybear/cronjobs` module to ensure my changes integrate well and to view the current state of testing.

**Command:**
`coverage run --source=greedybear/cronjobs manage.py test tests.greedybear.cronjobs tests.test_mass_scanners --keepdb`

**Results:**

| File | Coverage | Status |
| :--- | :--- | :--- |
| **`greedybear/cronjobs/cleanup.py`** | **100%** | **This PR** |
| `greedybear/cronjobs/monitor_honeypots.py` | 100% | Full coverage |
| `greedybear/cronjobs/monitor_logs.py` | 100% | Full coverage |
| `greedybear/cronjobs/mass_scanners.py` | 97% | High coverage |
| `greedybear/cronjobs/firehol.py` | 94% | High coverage |
| `greedybear/cronjobs/extract.py` | 0% | Needs attention (Out of scope) |
| `greedybear/cronjobs/whatsmyip.py` | 0% | Needs attention (Out of scope) |

**Comment by regulartim** [comment]
Thanks for your coverage check. I'll open an issue for testing `greedybear/cronjobs/whatsmyip.py`.

File: tests/greedybear/cronjobs/test_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        # Mock the logger to verify logging calls
+        cleanup_job.log = MagicMock()
+
+        # Execute the run method
+        cleanup_job.run()
+
+        # Verify interactions with IocRepository
+        ioc_repo.delete_old_iocs.assert_called_once()
+        expected_ioc_date = datetime.now() - timedelta(days=IOC_RETENTION)
```
Comment (regulartim) [inline tests/greedybear/cronjobs/test_cleanup.py]: I don't like that we are depending on an environment variable here. If, for example, I have it set to a value other than the default one, the test will probably fail on my machine.

Comment (opbot-xd) [reply tests/greedybear/cronjobs/test_cleanup.py]: Thanks for catching this! You're absolutely right - relying on environment variables makes the tests non-deterministic and brittle.
I should mock the retention settings to make the tests independent of the environment configuration. I'll update the tests to use `@patch` to override `IOC_RETENTION`, `COWRIE_SESSION_RETENTION`, and `COMMAND_SEQUENCE_RETENTION` with fixed values.
This way, the tests will verify the logic (i.e., "does it calculate the correct date from the setting?") without depending on what's actually configured on the machine.
Let me push a fix for this.

File: tests/greedybear/cronjobs/test_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+from datetime import datetime, timedelta
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from greedybear.cronjobs.cleanup import CleanUp
+from greedybear.cronjobs.repositories import CowrieSessionRepository, IocRepository
+
+
+class TestCleanUp(TestCase):
```
[the remark this answers is not in the document]
Comment (regulartim) [reply tests/greedybear/cronjobs/test_cleanup.py]: We have a separate issue for that.

[the remark this answers is not in the document]
Comment (regulartim) [reply tests/greedybear/cronjobs/test_cleanup.py]: Same here.

File: tests/greedybear/cronjobs/test_cleanup.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        # Execute the run method
+        cleanup_job.run()
+
+        # Verify interactions with IocRepository
+        ioc_repo.delete_old_iocs.assert_called_once()
+        expected_ioc_date = datetime.now() - timedelta(days=100)
+        # Check that the date passed is approximately correct (within 1 second)
+        args, _ = ioc_repo.delete_old_iocs.call_args
+        self.assertAlmostEqual(args[0], expected_ioc_date, delta=timedelta(seconds=1))
```
[the remark this answers is not in the document]
Comment (regulartim) [reply tests/greedybear/cronjobs/test_cleanup.py]: What do you think of that @opbot-xd ?

[the remark this answers is not in the document]
Comment (opbot-xd) [reply tests/greedybear/cronjobs/test_cleanup.py]: Great catch! I completely agree with Copilot's assessment. I've updated the tests to manually calculate the time difference using `.total_seconds()` and then assert that it's within the acceptable range using `assertLess`.

This approach is more explicit and provides clearer error messages if the assertion fails. The updated implementation now shows exactly how many seconds the difference is, making debugging much easier.

All tests still pass with the new approach. ✅

**Review by regulartim — APPROVED** [review]
